consensus/src/twins_test.rs

```diff
@@ -353,18 +403,174 @@ fn twins_vote_dedup_test() {
         // Pull enough votes to get a few commits.
         // The proposer's votes are implicit and do not go in the queue.
         playground
-            .wait_for_messages(30, NetworkPlayground::votes_only)
+            .wait_for_messages(50, NetworkPlayground::votes_only)
             .await;
 
         // No node should be able to commit because of the way partitions
         // have been created
         let mut commit_seen = false;
         for node in &mut nodes {
-            node.commit_cb_receiver.close();
             if let Ok(Some(_node_commit)) = node.commit_cb_receiver.try_next() {
                 commit_seen = true;
             }
         }
         assert!(!commit_seen);
     });
 }
+
+#[test]
+/// This test checks that when a node becomes a proposer, its
+/// twin becomes one too.
+///
+/// Setup:
+///
+/// 4 honest nodes (n0, n1, n2, n3), and 2 twins (twin0, twin1)
+/// Create 2 partitions, p1=[n0, n1, n2], p2=[n3, twin0, twin1]
+/// Let n0 (and implicitly twin0) be proposers
+///
+/// Test:
+///
+/// Extract enough votes so nodes in both partitions form commits.
+/// The commits should be on two different blocks
+///
+/// Run the test:
+/// cargo xtest -p consensus twins_proposer_test -- --nocapture
+#[cfg(test)]
```

Comment:
remove this line

Reply:
Done (assuming you meant line#438 '#[cfg(test)]')